### DISCUSSION TITLE
Where do we put the design documents?

### DISCUSSION THREAD

**Discussion by jackwener** [vesoft-inc/nebula#3404] [General]
If a PR has a corresponding design document, where should we put the document publicly?

**Comment by Shylock-Hg** [comment]
Better to be public.

**Comment by jackwener** [comment]
> If a PR has a corresponding design document, where should we put the document publicly?

Google doc, or other option ?

**Reply by Shylock-Hg** [reply]
https://github.com/vesoft-inc/nebula/tree/master/docs/rfcs

**Reply by jackwener** [reply]
I don't think that it's a good idea that put doc for a PR in dir of repo.

**Reply by Shylock-Hg** [reply]
What's the problem?

**Reply by jackwener** [reply]
It means that we must put the doc inside PR. The doc sometime just record code flow, and it isn't formal (sometime in Chinese)

**Reply by wey-gu** [reply]
How about WIP documented in google doc or a discussion/issue? and finally moved to https://github.com/vesoft-inc/nebula/tree/master/docs/rfcs ?

**Reply by wey-gu** [reply]
And dev docs could be rendered from https://github.com/vesoft-inc/nebula/tree/master/docs/rfcs easily with different github tags

**Reply by Shylock-Hg** [reply]
Put it in PR is better to be reviewed.

**Comment by wey-gu** [comment]
Or in the design flow, we firstly review and merge the [RFCs](https://github.com/vesoft-inc/nebula/tree/master/docs/rfcs ) PR before reviewing the implementation PR.

There could be a lite version of RFCs in issues/google docs(should be easier to be accessed and long-lived) before/during the RFC PR

And dev docs could be rendered from https://github.com/vesoft-inc/nebula/tree/master/docs/rfcs easily with different git tags

**Reply by Shylock-Hg** [reply]
It's great.

**Comment by wey-gu** [comment]
TiKV puts RFCs in a repo: https://github.com/tikv/rfcs

**Comment by wey-gu** [comment]
A short but clear post by @xuanwo https://xuanwo.io/reports/2021-26/
